products/statement-generator/src/__tests__/routing.test.tsx

```diff
@@ -156,24 +177,36 @@ describe('Routing Tests', () => {
     fireEvent.change(whyClear, { target: { value: 'The why' } });
     fireEvent.change(clearHow, { target: { value: 'The how' } });
 
-    const whyButton = screen.getByText(/next/i);
-    fireEvent.click(whyButton);
+    // Why -> Why Preview
+    fireEvent.click(screen.getByText(/next/i));
     expect(screen.getByTestId('location-display')).toHaveTextContent(
       '/why/preview'
     );
 
-    // tests why preview navigates to finalize
-    const whyPreviewButton = screen.getByText(/next/i);
-    fireEvent.click(whyPreviewButton);
+    // Why Preview -> Modal -> Finalize
+    fireEvent.click(screen.getByText(/next/i));
+    expect(screen.getByRole('dialog')).toBeInTheDocument();
+    expect(screen.getByTestId('location-display')).toHaveTextContent(
+      '/why/preview'
+    );
+
+    confirmModalAndFFW();
     expect(screen.getByTestId('location-display')).toHaveTextContent(
       '/finalize'
     );
 
-    // tests finalize navigates to finalize preview
-    const finalizeButton = screen.getByText(/next/i);
-    fireEvent.click(finalizeButton);
+    // Finalize -> Finalize Preview
+    fireEvent.click(screen.getByText(/next/i));
```

Comment:
<img width="1379" height="480" alt="Image" src="https://github.com/user-attachments/assets/3933bec5-c511-4106-aa9d-50b5f8bf7a2e" />

All the tests are passing, but I'm getting this console.error

Reply:
See if you can figure this out, but if its not a quick thing, write a new issue to fix it. 

I normally don't love to do that, but I just noticed that the routing tests in `dev` are a _mess_ and you fixed those with this update. 

Thank you so much for thinking about the tests ✨

Reply:
Beautiful! In which case, I'll resolve this and get it merged. Thank you so much!